### PR TITLE
Feature/docs architecture

### PR DESCRIPTION
**Pull request by kareziko**
Add documentation

### CODE REVIEW

File: docs/quantum_oscilator/architecture.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Architecture of Neutal Network
+
+## Parent classes
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:3]: Parent class of what? There is no class signature anywhere, please add it

File: docs/quantum_oscilator/architecture.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+[`#!python class tf.keras.Model`](https://keras.io/api/models/model/)
+
+## Introduction
+
+QONetwork class is responsible for the learning process from the user input.
+Our neural network architecture that simultaneously learns eigenvalues and the
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:10]: I have the impression that this sentence does not make much sense, please correct it - it is not architecture that learns eigenvalues etc.

File: docs/quantum_oscilator/architecture.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+### Abstract
+
+The construction of the network involves the definition of layers in the
+assemble_hook method. In neural networks there is one input layer (input), one
+output layer (output) and many hidden layers (d1, d2, ...). Adding hidden
+layers involves adding consecutive lines of code in the form dX =
+keras.layers.Dense(number_neurons, activation function, layer_name, number
+precision)(layer_previous).
+
+When building a neural network architecture, parameters within each layer can
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:35]: It's not true in this case. Our network has pre defined activation functions. Only number of neurons in internal layer can be changes with QOConstants

File: docs/quantum_oscilator/architecture.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+be manipulated, e.g.: different activation functions (sin, relu, tanh), number
+of neurons in each hidden layer, and changing the precision of the data.
+
+## Instance attributes
+
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:40]: I would prefer the attribute descriptions to keep the same format as for the QOParams class

File: docs/quantum_oscilator/architecture.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+| name            | description                             |
+| --------------- | --------------------------------------- |
+| constants       | constant values used in neutral network |
+| is_debug        |                                         |
+| loss_function   | loss function of neutral network        |
+| is_console_mode |                                         |
+
+## Constructor
+
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:49]: Change contructor to function signature and include parameters and returns table similar to how QOParams methods were described

File: docs/quantum_oscilator/architecture.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+Function assemble_hook is called first, followed by the constructor of the
+parent class [`tf.keras.model`](https://keras.io/api/models/model/). The
+get_loss_function is also called in the constructor. Worth to mention
+`super().__init__(inputs=inputs, outputs=outputs)` <-- This values are
+necessary to construct [`tf.keras.model`](https://keras.io/api/models/model/),
+and they are used as input and output layers of neutral network for `tf.keras`
+to decode structure of neutral network.
+
+## Assemble Hook
+
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:59]: Include function signature and description of the function (again, similar to QOParams). You may consider simplifying descriptions of how this function work written above, to just simply saying that it creates network structure, and provide an in-depth description here instead.

File: docs/quantum_oscilator/architecture.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+### Returns
+
+| type                   | description |
+| ---------------------- | ----------- |
+| `Layers (input layer)` |             |
+| `Layers (dense)`       |             | 
+
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:66]: Please add descriptions and make sure types match with types in the function signature.

File: docs/quantum_oscilator/architecture.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    B --> C((QQNetwork))
+    C --> D[Train_generation]
+    D --> E[train]
+    E-->C
+    F((QQParams)) --> D
+```
+
+```
+
+```
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:81]: Please remove this redundant code block

File: docs/quantum_oscilator/architecture.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+### Returns
+
+| type                   | description |
+| ---------------------- | ----------- |
+| `Layers (input layer)` |             |
+| `Layers (dense)`       |             | 
+
+## Graph of architecture
+
```
Comment (Argmaster) [inline docs/quantum_oscilator/architecture.md:68]: Graph is cool, but change Train_generation to lowercase and add parentheses at the end to indicate its a function (same applies to train). Change loop QONetwork->train_generations->train->QONetwork to train_generations->train->train_generations, no connection between train and QONetwork

**Review by Argmaster — COMMENTED** [review]
Please fix issues pointed out in review

File: docs/quantum_oscilator/architecture.md (hunk 1 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:3]: Nie rozumiem jest tak samo jak u kulisia, mówiliśmy na spotkaniu we wtorek że parent klas co keras.model

File: docs/quantum_oscilator/architecture.md (hunk 4 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:40]: chodzi o to żeby były po myślniku?
⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋⤋
## Instance attributes

    - constants --> constant values used in neutral network
    - is_debug 
    - loss_function --> loss function of neutral network
    - is_console_mode

File: docs/quantum_oscilator/architecture.md (hunk 5 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:49]: 1. Co to sygnatura funkcji?
2. Gdzie ty masz te tabelki w QQParams?????????

File: docs/quantum_oscilator/architecture.md (hunk 6 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:59]: To samo co wyżej

File: docs/quantum_oscilator/architecture.md (hunk 9 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:68]: Nawiasów nie można dodawać w grafach bo się kaszani.

File: docs/quantum_oscilator/architecture.md (hunk 1 above)
Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:3]: Nie rozumiem jest tak samo jak u kulisia, mówiliśmy na spotkaniu we wtorek że parent class to keras.model*

Comment (Argmaster) [reply docs/quantum_oscilator/architecture.md:3]: Chodzi o to, że nigdzie nie pojawia się sygnatura klasy, tj nagłówek class QONetwork, zobacz na branchu feature/docs-params jak wygląda plik params.md

Comment (kareziko) [reply docs/quantum_oscilator/architecture.md:3]: Niżej jest więcej pytań, proszę zerknij

File: docs/quantum_oscilator/architecture.md (hunk 5 above)
Comment (Argmaster) [reply docs/quantum_oscilator/architecture.md:49]: 1. https://developer.mozilla.org/en-US/docs/Glossary/Signature/Function

2. My bad, what i ment is this thing:
```
### `#!python def push_stats(self, ...) -> None`

Called after each learning epoch of the network. Adds new values to the history
of learning process metrics values.

#### Parameters

Each parameter will be added to the corresponding class attribute (see names).

| name              | type             |
| ----------------- | ---------------- |
| `total_loss`      | `#!python float` |
| `eigenvalue`      | `#!python float` |
| `residuum`        | `#!python float` |
| `function_loss`   | `#!python float` |
| `eigenvalue_loss` | `#!python float` |
| `drive_loss`      | `#!python float` |
| `c`               | `#!python float` |

#### Returns

| type   | description                    |
| ------ | ------------------------------ |
| Figure | Figure used for drawing plots. | 

```

File: docs/quantum_oscilator/architecture.md (hunk 4 above)
Comment (Argmaster) [reply docs/quantum_oscilator/architecture.md:40]: https://github.com/Argmaster/NNEVE/blob/feature/docs-params/docs/quantum_oscilator/params.md#L14-L25

File: docs/quantum_oscilator/architecture.md (hunk 9 above)
Comment (Argmaster) [reply docs/quantum_oscilator/architecture.md:68]: Ok, that's true, also my bad

**Comment by Argmaster** [comment]
Why have you force pushed all of your changes? In case of such non critical changes in source should create new commit. Renames or critical changes which can not be kept in version control are only examples of things to force push. However, don't change it now, it's only a note for the future.

**Comment by Argmaster** [comment]
I believe you are going to fill missing descriptions of class attributes?

**Comment by kareziko** [comment]
To jest wszystko co z dokumentacji potrafiłem "wydukać". Wybacz ale dalej nie wiem co mam wpisywać.

**Pull request closed without merging**